### ISSUE TITLE
convergence of in-the-loop training

### ISSUE THREAD

**Issue by iammarvelous** [nkolot/SPIN#42]
Hi, I wanted to reproduce the in-the-loop training, but it seems that the loss becomes extremely large after about three epochs. I use the following parameters
```
python train.py --name train_example --run_smplify --batch_size 160  --num_smplify_iters 50  --lr 1e-4
```
The only thing that's different is the batch size and learning rate, I double the learning rate since the batch size is almost doubled. I am trying to use the default learning rate, but I don't think this is the issue.
Did you encounter this in your experiment? Also, for the static training (no `--run_smplify`), I found `loss_regr_betas` also gets some extreme values, something like the first half of the `loss_regr_betas` in the following figures. Do you have any idea why these happen? 

![image](https://user-images.githubusercontent.com/14206232/72003416-55013300-3217-11ea-9c73-76da226846b0.png)
![image](https://user-images.githubusercontent.com/14206232/72003563-a3163680-3217-11ea-8f02-63e4e5367ff2.png)
![image](https://user-images.githubusercontent.com/14206232/72003672-e07ac400-3217-11ea-9bf9-7f541ecc46a8.png)
![image](https://user-images.githubusercontent.com/14206232/72003728-f9837500-3217-11ea-90bb-f74e9ad657ea.png)

**Comment by nkolot** [comment]
Hmm it looks like a learning rate issue. Can you try and train with a 10 times lower learning rate? Unfortunately Adam does not behave like SGD, so scaling the learning rate when increasing the batch size can have unexpected behavior.

**Comment by iammarvelous** [comment]
OK. I will try that. I tried static training before and everything looks good except that `loss_regr_betas` has some extreme values once a while.  I am curious about your training details. Did you train without `run_smplify` at the beginning? Also, did you use the default loss weight [here](https://github.com/nkolot/SPIN/blob/master/utils/train_options.py)? It seems that `shape_loss_weight` is 0? Also about `openpose_train_weight` and `gt_train_weight`, any suggest on this? Do you also suggest `openpose_train_weight` as in the [fitting stage](https://github.com/nkolot/SPIN/issues/30#issuecomment-570276278)?

**Comment by nkolot** [comment]
The parameters that you mentioned are the ones used in the paper. I also inspected the code and I noticed that there is a small bug in the released version. Essentially I am not filtering out extreme betas in the static fits, only for the in the loop fits. I will do the modification now and I will push the changes.

**Comment by iammarvelous** [comment]
I think this should be the reason for the bumps in `loss_regr_betas`. I recalled in the paper you mentioned about the per vertices loss, i.e., shape loss in the code. Did you ever run the experiment on the shape loss? If so, did you disable it because of inferior performance?

**Comment by nkolot** [comment]
We mention in the supplementary material that the weight we chose for this loss is 0. The reason that we ended up not using this loss is that it is highly correlated with the other losses.

**Comment by iammarvelous** [comment]
Well, I just take a look at your commit b4c70d5. The above figures are run under `run_smplify`. So I feel there may be other reasons for the bumps?

**Comment by nkolot** [comment]
So just to be clear, the previous version of the code could still be using the extreme betas from the static fits. What I do is I check which fit has the lowest reprojection error and in the beginning of training it is almost certain that the static fits have lower error because the network has not started producing accurate shapes, i.e. good initializations for the optimization. This is one reason we enable the in-the-loop training later on.

**Comment by ikvision** [comment]
@iammarvelous were you able to avoid this problem by reducing the learning_rate or enlarging the batch_size?
